docs/19_feature_design_template.md

```diff
@@ -1,45 +1,96 @@
 # Feature design
-
 > Feature description / User story.
 
+## Contents
+
+In this section, provide the contents for the document including all custom subsections created.
+
+- [**Analysis**](#analysis)
+    - [Feasibility study](#feasibility-study)
+    - [Requirements](#requirements)
+    - [Landscape](#landscape)
+    - [Prototyping](#prototyping)
+    - [System modeling](#system-modeling)
+- [**Design**](#design)
+    - [Architecture](#architecture)
+    - [User Interface](#user-interface)
+    - [Database](#database)
+    - [Program](#program)
+
 # Analysis
+> Describe a general analysis approach.
 
-> Describe general analysis approach.
+In this section, describe the methodology to feature analysis, possible solutions, and areas that need to be explored, ways to compare alternatives. It should be a brief plan for all sub-sections.
 
-### Feasibility study
+_**Important Note**: Avoid excessive details in the Analysis subsections. Think about the feature at the very top-level abstraction. Provide the details only where necessary (e.g. in the [Prototyping](#prototyping) section)._
 
+### Feasibility study
 > A preliminary study of the feasibility of implementing this feature.
 
-### Requirements
+Once the requirements are defined, one should consider the feature and conclude if this feature makes sense for the project. In the context of usability and project's purposes, one should answer the following: 
+
+- How this feature would help end-users and would they use it?
+- Does this feature meet the project's scope?
+- Does this feature fit the project's purpose?
 
+Once the feature makes sense, one should review is it feasible. More precisely, one should conclude is it possible to implement the feature taking into account the listed requirements from the [Requirements](#requirements) section. It is possible here to refer to existing solutions in the [Landscape](#landscape) section.
```

Comment:
```suggestion
```

Let's skip references to the future sections, I think for now we can limit to the above questions.